### PR TITLE
Format and lint Markdown text files

### PR DESCRIPTION
**Pull request by jason-fox**
Similar to #377 - This change to package.json introduces markdown [textlint](https://github.com/textlint/textlint)  linting  and markdown formatting using prettier.

* The text is checked for spelling errors, bad grammar and dead links (on CI)
* The markdown is nicely formatted. (on Husky commit)
* The code snippets within the markdown are formatted (on Husky commit)

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,7819 @@
+{
```
Comment (fgalan) [inline package-lock.json]: Please remove package-lock.json from the PR.

File: package.json [hunk 2]
```diff
@@ -24,25 +24,35 @@
...
-    "nock": "10.0.1",
     "mocha": "5.2.0",
-    "should": "13.2.3",
-    "sinon": "~6.1.0",
-    "istanbul": "~0.4.5",
+    "nock": "10.0.1",
     "prettier": "~1.14.2",
     "proxyquire": "2.1.0",
+    "remark-cli": "~6.0.1",
+    "remark-preset-lint-recommended": "^3.0.2",
```
Comment (fgalan) [inline package.json]: ^ -> ~ in dependencies.

File: package-lock.json (hunk 1 above)
Comment (jason-fox) [reply package-lock.json]: Fixed d048ba6

File: package.json (hunk 2 above)
Comment (jason-fox) [reply package.json]: Fixed d048ba6

**Comment by dcalvoalonso** [comment]
LGTM

**Pull request merged**